docs/src/main/sphinx/connector/opensearch.md

```diff
@@ -0,0 +1,446 @@
+| `opensearch.aws.access-key`  | AWS access key to use to connect to the OpenSearch domain. If not set, the Default AWS Credentials Provider chain will be used.              |
+| `opensearch.aws.secret-key`  | AWS secret key to use to connect to the OpenSearch domain. If not set, the Default AWS Credentials Provider chain will be used.              |
+| `opensearch.aws.iam-role`    | Optional ARN of an IAM Role to assume to connect to the OpenSearch domain. Note: the configured IAM user has to be able to assume this role. |
+| `opensearch.aws.external-id` | Optional external ID to pass while assuming an AWS IAM Role.                                                                                 |
+
+## Password authentication
+
+To enable password authentication, the `opensearch.security` option needs to be set to `PASSWORD`.
+Additionally the following options need to be configured appropriately:
+
+| Property name              | Description                                |
+|----------------------------|--------------------------------------------|
+| `opensearch.auth.user`     | User name to use to connect to OpenSearch. |
+| `opensearch.auth.password` | Password to use to connect to OpenSearch.  |
+
+(opensearch-sql-support)=
+
+## SQL support
+
+The connector provides {ref}`globally available <sql-globally-available>` and
+{ref}`read operation <sql-read-operations>` statements to access data and
+metadata in the OpenSearch catalog.
+
+## Table functions
+
+The connector provides specific {doc}`table functions </functions/table>` to
+access OpenSearch.
+
+(opensearch-raw-query-function)=
+
+### `raw_query(varchar) -> table`
+
+The `raw_query` function allows you to query the underlying database directly.
+This function requires [OpenSearch Query DSL](https://opensearch.org/docs/latest/query-dsl/index/)
+syntax, because the full query is pushed down and processed in OpenSearch.
+This can be useful for accessing native features which are not available in
+Trino or for improving query performance in situations where running a query
+natively may be faster.
+
+```{include} query-passthrough-warning.fragment
+```
+
+The `raw_query` function requires three parameters:
+
+- `schema`: The schema in the catalog that the query is to be executed on.
+- `index`: The index in OpenSearch to be searched.
+- `query`: The query to be executed, written in Elastic Query DSL.
+
+Once executed, the query returns a single row containing the resulting JSON
+payload returned by OpenSearch.
+
+For example, query the `example` catalog and use the `raw_query` table
+function to search for documents in the `orders` index where the country name
+is `ALGERIA`:
+
+```
+SELECT
+  *
+FROM
+  TABLE(
+    example.system.raw_query(
+      schema => 'sales',
+      index => 'orders',
+      query => '{
+        "query": {
+          "match": {
+            "name": "ALGERIA"
+          }
+        }
+      }'
+    )
+  );
+```
+
```

Comment:
Optionally .. add a mention that the result can then be processed with the JSON function